x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/popover_editor.tsx

```diff
@@ -358,6 +351,7 @@ export function PopoverEditor(props: PopoverEditorProps) {
                     state={state}
                     setState={setState}
                     columnId={columnId}
+                    currentColumn={state.layers[layerId].columns[columnId]}
```

Comment:
My comment in the description about this was probably not super clear:

> Adds currentColumn to param editor props to avoid plugging it out of the state object and casting it in every param editor


It's because of types - `columns` is a `Record<string, IndexPatternColumn>`, but inside of the param editor we need the specific type, `TermsIndexPatternColumn`, `DateHistogramIndexPatternColumn` and so on. We can cast it in there (that's how it was done before), but I try to get rid of casts where possible because they are a type mismatch waiting to happen. Doing it here is also implicitly casting the type to right one, but it's only a single place compared to the n places within in the param editor components, so less surface to get it wrong in the future.